analyzers/src/SonarAnalyzer.CSharp/Rules/UseModelBinding.cs

```diff
@@ -0,0 +1,253 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2024 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Concurrent;
+
+namespace SonarAnalyzer.Rules.CSharp;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class UseModelBinding : SonarDiagnosticAnalyzer<SyntaxKind>
+{
+    private const string DiagnosticId = "S6932";
+    private const string UseModelBindingMessage = "Use model binding instead of accessing the raw request data";
+    private const string UseIFormFileBindingMessage = "Use IFormFile or IFormFileCollection binding instead";
+
+    protected override string MessageFormat => "{0}";
+
+    protected override ILanguageFacade<SyntaxKind> Language => CSharpFacade.Instance;
+
+    public UseModelBinding() : base(DiagnosticId) { }
+
+    protected override void Initialize(SonarAnalysisContext context)
+    {
+        context.RegisterCompilationStartAction(compilationStartContext =>
+        {
+            var (argumentDescriptors, propertyAccessDescriptors) = GetDescriptors(compilationStartContext.Compilation);
+            if (argumentDescriptors.Any() || propertyAccessDescriptors.Any())
+            {
+                compilationStartContext.RegisterSymbolStartAction(symbolStartContext =>
+                {
+                    // If the user overrides any action filters, model binding may not be working as expected. Then we do not want to raise on expressions that originate from parameters.
+                    // See the OverridesController.Undecidable test cases for details.
+                    var hasOverrides = false;
+                    var controllerCandidates = new ConcurrentStack<ReportCandidate>(); // In SymbolEnd, we filter the candidates based on the overriding we learn on the go.
+                    if (symbolStartContext.Symbol is INamedTypeSymbol namedType && namedType.IsControllerType())
+                    {
+                        symbolStartContext.RegisterCodeBlockStartAction<SyntaxKind>(codeBlockStart =>
```

Comment:
Without suffix works well for "Start" actions, but for other context like "NodeContext" you need the suffix. I removed the suffix from Start/End actions and kept it otherwise.